test/integration/plan_test.exs

```diff
@@ -13,4 +13,39 @@ defmodule Braintree.Integration.PlanTest do
     assert plan.id
     assert plan.name
   end
+
+  # This will create a new plan every time it's ran, make sure to cleanup if ran excessively
```

Comment:
Ideally we'd delete them afterwards to clean up. I'm not sure what the plan limit is, and I'm sure we don't want to clean them out manually.

Reply:
I agree, ideally I can delete and clear them out manually as well. (Would also be helpful for preloading data for testing `Plan.all()`) 

But it seems that a DELETE request on "plans/#{plan_id}" works properly but returns html due to it probably being something expected to be used from within the web platform, which breaks the XML decoder. I included some code below to show what I mean:
```
iex(2)> Braintree.HTTP.delete("plans/p8bg")
"<html><body>You are being <a href=\"https://api.sandbox.braintreegateway.com/merchants/2ww49grjxnbgqwb6/plans\">redirected</a>.</body></html>"
{:ok, :ok}
[error] unprocessable response
```

Maybe we just add a private function in `plan_test.exs` to handle deleting post and ignore the failing decoding?

Reply:
Ah, they redirect after a delete request. The response should have a 30X response code. Can you modify the HTTP client to handle redirects differently? Don't bother parsing the response and return an `:ok` tuple or empty binary instead?